typescript/provider-sdk/src/ism.ts

```diff
@@ -1,11 +1,21 @@
-import { WithAddress } from '@hyperlane-xyz/utils';
+import { WithAddress, deepEquals } from '@hyperlane-xyz/utils';
+
+import { ExtractModuleType } from './module.js';
 
 export type IsmModuleType = {
   config: IsmConfig;
   derived: DerivedIsmConfig;
   addresses: IsmModuleAddresses;
 };
 
+/**
+ * Helper type for ISM module types.
+ */
+export type ExtractIsmModuleType<TType extends IsmType> = ExtractModuleType<
```

Comment:
Together with `ExtractModuleType` this feels too complex? I'd prefer making `IsmModuleType` generic or using the approach we take with `IsmConfigs` where you can do  `IsmConfigs['typeA' | 'typeB']`